download/config.go

```diff
@@ -14,6 +14,14 @@ const (
 	defaultMaxDelaySeconds = int64(120)
 )
 
+var (
+	// TriggerPeriodic represents periodic polling mechanism
```

Comment:
nit: It might be worth adding a `DefaultTriggerMode` constant that refers to the periodic one and then using that throughout. I noticed we're duplicating the default value in the discovery plugin and that spurred the thought.

Reply:
Done